src/spec/template_spec.js

```diff
@@ -0,0 +1,333 @@
+        });
+          
+        $httpBackend.whenGET(/\/accounts\/[\w|-]*\/status\/limits/).respond(200, {
+          "noLimits": true,
+          "dkimConfigurationRequired": false,
+          "domainConfigurationRequired": false,
+          "_links": []
+        });
+      }
+    ));
+  }
+
+  function setupCreateTemplateSuccessResponse() {
+    browser.addMockModule('descartableModule3', () => angular
+    .module('descartableModule3', ['ngMockE2E'])
+    .run($httpBackend => {
+
+      $httpBackend.whenGET(/\/accounts\/[\d|-]*\/templates/).respond(200, {
+        "items": [
+          {
+            "from_name": "test name",
+            "from_email": "test@test.test",
+            "subject": "test subject",
+            "body": "Template body test",
+            "name": "Test template",
+            "id": "1169ab26-78dc-4474-a8c2-e1f02f6c05ed",
+            "last_updated": "2018-07-12T15:45:19Z",
+            "created_at": "2018-07-12T13:27:40Z",
+            "bodyType": "rawHtml",
+            "_links": []
+          }
+        ],
+        "pageSize": 200,
+        "itemsCount": 1,
+        "currentPage": 1,
+        "pagesCount": 1,
+        "_links": []
+      });
+
+      $httpBackend.whenPUT(/\/accounts\/[\d|-]*\/templates\/[\w|-]*\/body/).respond(200, {
+        "message": "Template successfully updated",
+        "_links": []
+      });
+
+      $httpBackend.whenPOST(/\/accounts\/[\d|-]*\/templates/).respond(200, {
+        "createdResourceId": "1169ab26-78dc-4474-a8c2-e1f02f6c05ed",
+        "message": "Template successfully created",
+        "_links": []
+      });
+    }));
+  }
+
+  function setupCreateNewTemplateFailResponse() {
+    browser.addMockModule('descartableModule4', () => angular
+    .module('descartableModule4', ['ngMockE2E'])
+    .run($httpBackend => {
+
+      $httpBackend.whenGET(/\/accounts\/[\d|-]*\/templates/).respond(200, {
+        "items": [
+          {
+            "from_name": "test name",
+            "from_email": "test@test.test",
+            "subject": "test subject",
+            "body": "Template body test",
+            "name": "Test template",
+            "id": "1169ab26-78dc-4474-a8c2-e1f02f6c05ed",
+            "last_updated": "2018-07-12T15:45:19Z",
+            "created_at": "2018-07-12T13:27:40Z",
+            "bodyType": "rawHtml",
+            "_links": []
+          }
+        ],
+        "pageSize": 200,
+        "itemsCount": 1,
+        "currentPage": 1,
+        "pagesCount": 1,
+        "_links": []
+      });
+
+      $httpBackend.whenPUT(/\/accounts\/[\d|-]*\/templates\/[\w|-]*\/body/).respond(200, {
+        "message": "Template successfully updated",
+        "_links": []
+      });
+
+      $httpBackend.whenPOST(/\/accounts\/[\d|-]*\/templates/).respond(500, {
+        "createdResourceId": "1169ab26-78dc-4474-a8c2-e1f02f6c05ed",
+        "message": "Template creation fail",
+        "_links": []
+      });
+    }));
+  }
+
+  function setupTemplateMenuSuccessResponse() {
```

Comment:
Menu?

Reply:
The list of templates, yeah is really ugly name, I will change it 👍